metadata-ingestion/src/datahub/ingestion/api/source.py

```diff
@@ -181,10 +182,32 @@ def get_workunit_processors(self) -> List[Optional[MetadataWorkUnitProcessor]]:
         """A list of functions that transforms the workunits produced by this source.
         Run in order, first in list is applied first. Be careful with order when overriding.
         """
+        browse_path_processor: Optional[MetadataWorkUnitProcessor] = None
+        if (
+            self.ctx.pipeline_config
+            and self.ctx.pipeline_config.flags
+            and self.ctx.pipeline_config.flags.get("generate_browse_path_v2")
+        ):
+            platform = getattr(self, "platform", None) or getattr(
+                self.get_config(), "platform", None
```

Comment:
does this handle self.get_config() returning none?

Reply:
I'd really like if `get_config` would be implemented for all sources. I wanted to write a test that would check that, but it was wayyyy too difficult because we put a lot of code in our init functions, so I can't create each source with generic constructor params.

I guess this gets auto applied to third party sources we don't have control over, so I'll change, but would like to soon make `get_config` an abstractmethod so we can actually rely on it.